modin/engines/base/frame/data.py

```diff
@@ -1662,10 +1662,15 @@ def broadcast_apply_full_axis(
         -------
              A new Modin DataFrame
         """
+        if other is not None and not isinstance(other, list):
+            other = [other]
+        other = (
+            [o._partitions for o in other] if other is not None and len(other) else None
```

Comment:
`[o._partitions for o in other if o is not None] if len(other) else None`

Reply:
it's not supposed that any of the value in a list will be `None`

Reply:
okay, changed it

Reply:
`other` could be:
- ModinFrame
- List of ModinFrames
- None

In the first case, we're wrapping the frame in a list and processing it with no problems.

In the second case, if we met `None` in a list of ModinFrames then it means that something bad happened, and we want to crash our code, to prevent undefined behavior.

In the third case `None` will be passed `broadcast_axis_partitions` and `other` parameter will be ignored.

I don't see cases when we need that check for `None` inside that for-loop:
```python
[o._partitions for o in other if o is not None]
```

Reply:
I don't mind on this, we can leave `[o._partitions for o in other] len(other) else None`. Moreover, we can leave just `[o._partitions for o in other]` as values in a list are supposed to be not `None`.

Reply:
I thought we agreed about the current approach...

I'm not sure that I understand your last comment, I think that all possible cases of `other` are described [here](https://github.com/modin-project/modin/pull/2461#discussion_r531037123), and what this comment says are matches with your last comment?

Reply:
I mean we can leave either as follows
```python
other = [o._partitions for o in other] if len(other) else None
# as `if other is not None` is checked on 1666 line
```
or as follows
```python
other = [o._partitions for o in other]
# as values in a list are not supposed to be None.
```

Reply:
okay, got it, changed it to at the last commit:
```python
if other is not None:
    if not isinstance(other, list):
        other = [other]
    other = [o._partitions for o in other] if len(other) else None
```